src/notification.test.ts

```diff
@@ -437,7 +478,43 @@ describe("BaseNotification", () => {
       expect(notification.fetchProtocolNegotiationInfo).toHaveBeenCalled();
     });
 
-    test("throws a FetchError if the negotiation connection info fetch fails", async () => {
+    test("throws a BadRequestError if the negotiation connection info fetch fails, using problem details", async () => {
+      const fetchFn = mockedFetchWithError(
+        400,
+        `{"status": 400, "title": "Connection info error", "detail": "Connection info error detail"}`,
+        { "Content-Type": "application/problem+json" },
+      );
+
+      const endpoint = "https://fake.url/some-endpoint";
+      const topic = "https://fake.url/some-resource";
+      const protocol = ["ws"] as Array<protocols>;
+      const notification = new BaseNotification(topic, protocol, {
+        fetch: fetchFn,
+      });
+
+      notification.fetchProtocolNegotiationInfo = jest
+        .fn(notification.fetchProtocolNegotiationInfo)
+        .mockResolvedValue({
+          endpoint,
+          protocol: "ws",
+          features: {},
+        });
+
+      const err = await notification
+        .fetchNotificationConnectionInfo()
+        .catch((e) => e as BadRequestError);
```

Comment:
Ditto on the type declaration

Reply:
Addressed in https://github.com/inrupt/solid-client-notifications-js/pull/1031/commits/700073b7c4ab7d505b9a959a7cc00dafb5d85e37